x-pack/plugins/xpack_main/server/lib/telemetry/monitoring/get_beats_stats.js

```diff
@@ -97,6 +102,15 @@ export function processResults(results = [], { clusters, clusterHostSets, cluste
         clusters[clusterUuid].module.names = Array.from(moduleSet);
         clusters[clusterUuid].module.count += stateModule.count;
       }
+
+      const stateHost = get(hit, '_source.beats_state.state.host');
+      if (stateHost !== undefined) {
+        const hostSet = clusterArchitectureSets[clusterUuid];
+        const hostArchPlatform = `${stateHost.architecture}/${stateHost.os.platform}`;
+        hostSet.add(hostArchPlatform);
+        clusters[clusterUuid].architecture.names = Array.from(hostSet);
```

Comment:
@pickypg So we'd store `architecture.architectures` and `architecture.platform` or something here? Going with your comment https://github.com/pickypg

Reply:
Actually, I'm not sure I'm following your recommendation here. Can you provide some psuedo code?

Reply:
```js
// set up top
const clusterArchitectureMaps = new Map();

// looping across hosts
if (stateHost !== undefined) {
  const hostMap = clusterArchitectureMaps[clusterUuid];
  const hostKey = `${stateHost.architecture}/${stateHost.os.platform}`;
  let os = hostMap.get(hostKey);

  if (!os) { // undefined if new
    os = { name: stateHost.os.platform, architecture: stateHost.architecture, count: 0 };
    hostMap.set(hostKey, os);
  }

  // total per os/arch
  os.count += 1;

  // overall total (which should be the same number as the sum of all os.count values)
  clusters[clusterUuid].architecture.count += 1;
}
```

I feel like the serialization of the map should happen only at the end to avoid repeating that effort for every step of the loop. (This is currently where you're doing `Array.from(hostSet)`)